### PR TITLE
Nukes spacepods and associated jobs

### LINKED ISSUE #3672: Strange object teleporter lets you escape space pod
You hop out, and leave a debug message saying that the pilot escaped. Probably needs an exit() proc or something similar

### LINKED ISSUE #5467: Pai abilities
**Problem Description**:
things like the Security Records and the Atmosphere sensor dont appear to work.
PS. The door jack thing takes a few clicks to make work

**What did you expect to happen**: 
Security Records: I expected to click on a name and have something appear as it looks like you can click on a name.
Atmosphere Sensor: To be able to see what is in the atmosphere
**What happened instead**:
Security records only showed me alphabetized names, and when I tried to click on a name nothing happened.
Atmosphere sensor: Always loading, nothing changed.

**Why is this bad/What are the consequences:**
Cant be very useful to security and I cant tell if there are toxins in the atmosphere to warn my host.

**Steps to reproduce the problem**:
Download security records and atmoshpere sensor, then try to use them.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**:
The entire system seemed a little slow to the uptake, like when I tried door jacking I had to spam click the button in order to actually start the hack.

### LINKED ISSUE #5909: Pods can be deleted by stationary meaty ores
**Problem Description**:
Me and my pod was deleted when running into what looked like safe meat 

**What did you expect to happen**: 
The meat was completely stationary next to a hull breech. So it should have just been normal food.

**What happened instead**:
Both me and my pod was deleted, removing me from the round without any way back other then special spawns like npc, drone, golem, ext. 

**Why is this bad/What are the consequences:**
People in pods can be deleted without any warning for what looks like a normal item.

**Steps to reproduce the problem**:
This will be harder to replicate, but find a meaty ore, run into it with a pod, or possibly be hit with it, and it should delete the pod.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: It was a vamp round but that shouldn't have affected the meaty ore or my pod.

### LINKED ISSUE #6979: Bomb Guardian makes more sec pods
**Problem Description**:
When you make a bomb on a sec pod (or maybe all pods) it makes a duplicate pod for the duration of the bomb.

**What did you expect to happen**: 
Unless guardians are suppose to be super sekret mechanics, not sure this is intended.

**What happened instead**:
I expected the pod to be armed as a bomb..IT made a new pod to be the bomb.

**Why is this bad/What are the consequences:**
It can block off sec pods from windows.

**Steps to reproduce the problem**:
Buy a injector, pick bomb guardian, bomb a sec pod, boom, new sec pod.

### LINKED ISSUE #7086: Station-bounced radios don't work inside spacepods
**Problem Description**:
Station-bounced radios don't work inside spacepods, the pilot has to exit the pod to make use of it on any Z-level.

**What did you expect to happen**: 
The radio should be able to send and receive radio communications within a spacepod.

**What happened instead**:
The pilot spends a lot of time talking to themselves, missing out on whatever's happening on the station if they're out of radio range.

**Why is this bad/What are the consequences:**
Security pod pilots can't communicate from within their pod if comms are down for whatever reason, and the mechanic/anyone else has to brave the vaccuum to get their message out (assuming they lack a space suit)

**Steps to reproduce the problem**:
1) Acquire pod
2) Fly to Z-level outside of station radio range
3) Turn on the microphone and talk
4) Repeat outside the pod to make sure the radio works

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**:
I've experienced this on probably every gamemode, so it's probably not related to that or anything specific to one round. I also feel like I was able to use SBR's from within the pod at one point though I can't confirm that really. Probably not specific to any one type of pod as well, ie security, civilian, industrial, etc.

### LINKED ISSUE #8341: Spacepods have a habit of shooting themselves
**Problem Description**:
If you keep moving forward while firing a spacepod, or the server decides to be particularly laggy for a moment (if, say, mining decided to dump a million ores into the ORM all at once), then you stand a pretty good chance of shooting your own pod with its own weapons.

**What did you expect to happen**: 
Should not be possible for your front mounted weapons to be able to shoot yourself in the back.

**What happened instead**:
Able to shoot self in the back.

**Why is this bad/What are the consequences:**
A single volley from a basic mining laser, the weakest (to my knowledge) weapon a pod can have, puts a civilian pod in critical condition. two shots is enough to cause a meltdown and gib the pilot. Does the mechanic not get a bad enough rap already without their pods having a tendency to kill the user?

**Steps to reproduce the problem**:
Mount weapon on pod, continue flying while firing pod, you will eventually hit yourself. Especially if you add in some lag.

### LINKED ISSUE #9245: Parked space pods get moved when white ship moves
**Problem Description**:
Moving the white ship makes spacepods get moved into walls.

**What did you expect to happen**: 
The spacepods to remain where they are.

**What happened instead**:
they got stuck in walls

**Why is this bad/What are the consequences:**
You can still move out of the wall, but it is still bad.

**Steps to reproduce the problem**:
Park pod in podbay of white ship, move ship using the console.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: 
See after picture below
untitled

### LINKED ISSUE #9565: Space Pods Immune to Tesla
**Problem Description**:
The Tesla does nothing to space pods.

**What did you expect to happen**: 
The space pod to blow up.

**What happened instead**:
It can freely fly around in the tesla with no harm to it or it's pilot.

**Why is this bad/What are the consequences:**
Tesla supposed to zap and kill/dust people and blow up machines, can use space pod to not give a fuck.

**Steps to reproduce the problem**:
Fly a pod around a tesla.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Irrelevant

### LINKED ISSUE #10172: character floats as if in zero gravity under some pod-related circumstances
**Problem Description**:
When you enter a spacepod in gravity, go out into space, hop out and back in and then fly back to the station and get out, your character still 'floats'/jiggles up and down as if in space, even when affected by gravity

**What did you expect to happen**: 
The character to be displayed properly.

**What happened instead**:
It looks like I am in zero g

**Why is this bad/What are the consequences:**
It's a visual bug, but it's annoying.

**Steps to reproduce the problem**:
1. Enter spacepod, take pod into space.
2. exit pod in zero-g, notice you're now floating up and down. Re-enter your space pod
3. take spacepod back to station, leave spacepod
4. notice you're still floating up and down.

**When did the problem start happening**:
Has been like that for a while.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
I assume the issue is the 'floating in space' animation is only checked/removed when you cross into/out of gravity, but it isn't checked if you're in a pod.

I usually fix it by taking my pod back out into space, exiting it, entering gravity and then from gravity entering my pod and taking it back across.

### LINKED ISSUE #11701: Space pods ore box does not pick up ore
**Problem Description**:
Space pods do not pick up ore when installed with an orebox.

**What did you expect to happen**: 
Space pods with the ore box module are supposed to pick up ore when running over ore.

**What happened instead**:
It didn't do anything and left the ore on the floor.

**Why is this bad/What are the consequences:**
Space pods are sorta useless to mine with now.

**Steps to reproduce the problem**:
Drive a spacepod over ore, when equiped with an orebox module.

**When did the problem start happening**:
No idea when it started. I tried it today though on lavaland.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
N/A

### LINKED ISSUE #11702: Space pods do not attract mob hostility in lavaland
**Problem Description**:
Ranged Mobs do not appear to attack space pods.

**What did you expect to happen**: 
Mobs to attack the space pod.

**What happened instead**:
Had not gotten the chance to try it on a melee mob, but ranged mobs do not seem to target it, and act as if their injured with no agro-er.

**Why is this bad/What are the consequences:**
A pretty big exploit apparently, to be able to murder atleast some of the things in lavaland with immunity.

**Steps to reproduce the problem**:
Have the mechanic help you build a mining space pod in lavaland. Find out that it makes you safe from mobs. The only mob I tested it on was the hierophant. (I still don't know how that mob works, all I know is it fired purple tile- lines at me at the start, and then did nothing but glow purple as I shot it.)

**When did the problem start happening**:
Presumably this started when lavaland was introduced? I don't know if this happened before with mining mobs in the past.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
Secret round, Don't know what kind. Nothing else out of the ordinary.

### LINKED ISSUE #11703: Space pods shooting self(mining KA burst)
**Problem Description**:
If you move forward in any direction and fire a mining KA shot thing, You will hit your own pod rather then what you are moving towards.

**What did you expect to happen**: 
Being able to move forward and still fire.

**What happened instead**:
Shot my own pod, knocking a large hunt of integrity off of it.

**Why is this bad/What are the consequences:**
Makes pod useless for mining, and could easily kill you if you weren't aware.

**Steps to reproduce the problem**:
Install mining weapon system on pod. Drive forward while firing.

**When did the problem start happening**:
No idea when it started.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
I had a pod built on lavaland. I couldn't mine rocks without hitting myself, until I realized I have to stop moving to fire, lest I blow up my pod.

### LINKED ISSUE #11708: Poddoors on escape shuttle move each time it moves
**Problem Description**:
Each time the escape shuttle moves, any poddoors I placed on it get moved. A 4 tile door moved 3 tiles up, a 2 tile door moved 1 tile up each time the shuttle transitions a z level.

**What did you expect to happen**: 
for pod doors to stay where they are.

**What happened instead**:
they moved.

**Why is this bad/What are the consequences:**
I wanted to make a cool shuttle with a pod docking port so the sec pod pilot can get in there while guarding the shuttle from space.

**Steps to reproduce the problem**:
open map editor, place a pod door on the shuttle, start a round. Since the shuttle immediately moves to z2, jump to z2 and see the new door location.

**When did the problem start happening**:
No clue

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
The white ship has pod doors and those don't work, so I dunno what's special about the escape shuttle.

### LINKED ISSUE #12134: Invisible space pods
**Issue Description**:
Space pods when inside a ship that transitions to another Z level causes the pods to go invisible.

**What did you expect to happen**: 
For the space pod to not go invisible.

**What happened instead**:
The space pod lost its sprite.

**Why is this bad/What are the consequences**:
Invisible things are bad.

**Steps to reproduce the problem**:
Put space pod in ship like the white ship, send it to another Z level.  Pod goes invisible.

**When did the problem start happening**:
Within the past two weeks.

**Extra information**:

### LINKED ISSUE #12267: Haunted pod parts
**Issue Description**:

You can rotate pod parts as a ghost.

**What did you expect to happen**: 

I expected to not be able to, cause, y'know, ghosts generally aren't allowed to interfere with the living and stuff.

**What happened instead**:

I could actually rotate the pod parts.

**Why is this bad/What are the consequences**:

I mean, it's not that important, but i don't know, someone migth want to be a dick and just fuck around with the mechanic, which could be annoying, altough, i don't know why someone would do that, just wanted to report this bug.

**Steps to reproduce the problem**:

Have someone (or something) fabricate pod parts with the pod fabricator, and, as a ghost, you'll be able to rotate the parts.

**When did the problem start happening**:

Honestly, it could have always been there wihouth anyone noticing, mainly because one doesn't expect to rotate stuff as a ghost.

**Extra information**:

### LINKED ISSUE #12778: Blobs cannot attack space pods
**Issue Description**:

Blobs cannot manually attack space pods unless the pods bottom left corner of the pod is against a tendril.

**What did you expect to happen**: 

For blobs to be able to attack pods regardless of the direction it is being clicked on.

**What happened instead**:

Blobs were not able to manually attack the pods by clicking on them as it was saying the pod was out of reach.

**Why is this bad/What are the consequences**:

It allows for exploiting of pods against blobs, leaving the blob with no way to attack a pod back outside of spores.

**Steps to reproduce the problem**:

have a pod touch up against a blob on the right side or top side of the pod.  The blob will not be able to attack it.

**When did the problem start happening**:

Not sure, it could of been a long standing issue with pods that really never got noticed.

**Extra information**:

### LINKED ISSUE #13212: Unable to passively heal as diona in spacepod.
**Issue Description**:

When in a spacpod, diona are unable to passively heal.

**What did you expect to happen**: 

Non-critical damage was taken as a diona. Expected the damage to heal.

**What happened instead**:

No healing effect took place

**Why is this bad/What are the consequences**:

In truth, this bug is a mild inconvenience at best, as it requires both someone to be in a spacepod, take non lethal damage, and be a diona. 

**Steps to reproduce the problem**:

Enter a spacepod without space protection as a diona. enter space. Set pod setting to take atmosphere from surroundings. revert to internal environment before critical damage is sustained.

**When did the problem start happening**:

last successful test of bug was on 3/27/2020

**Extra information**:

HUD did not show lack of light as diona. activated flashlight was equipped, spacepod lights were activated.

### LINKED ISSUE #14046: Held Mice in Space Pod die of Suffocation
**Issue Description**:

If someone holds a mouse and pilots a space pod into space, the mouse suffocates to death.

**What did you expect to happen**: 

The mouse would breathe the space pod's oxygen.

**What happened instead**:

The mouse suffocated to death.

**Why is this bad/What are the consequences**:

It kills mice carried onto the spacepod when it shouldn't.

**Steps to reproduce the problem**:

1. Grab a mouse.
2. Enter space pod.
3. Fly around space until mouse suffocates to death.

**When did the problem start happening**:

**Extra information**:

### LINKED ISSUE #16088: Spacepod code has a different subtype for hearing messages
```dm
/obj/spacepod/hear_talk/hear_talk(mob/M, list/message_pieces)
	cargo_hold.hear_talk(M, message_pieces)
	..()

/obj/spacepod/hear_message(mob/M, msg)
	cargo_hold.hear_message(M, msg)
	..()
```

Why.

Good first issue if someone wants it .

This makes a new type for no reason
image

### LINKED ISSUE #16734: Remote View genetics power doesn't work while inside a spacepod
**Issue Description**:
Can't spy on people from the depths of space

**What did you expect to happen**: 
To be able to sit in my comfy pod and look through peoples' eyes

**What happened instead**:
My view didn't change, and the power went on cooldown. Worked fine once I stepped out of the pod.

**Steps to reproduce the problem**:
Get the genetics remote viewing power, climb inside a spacepod, try to use the power

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Nukes spacepods and their associated jobs. Anyone who had bought these jobs is eligible for a karma refund, and can be done from within the karma shop on the refunds page
image

Closes #3672
Closes #5467
Closes #7086
Closes #8341
Closes #9245
Closes #10172
Closes #11701
Closes #13212
Closes #14046
Closes #16088
Closes #16734
Closes #12778
Closes #6979
Closes #5909
Closes #9565
Closes #11702
Closes #11703
Closes #11708
Closes #12134
Closes #12267

## Why It's Good For The Game
This was discussed in the community meeting and within leadership. Of the 6 out of 7 people who voted, all 6 of them voted in favour of removing spacepods. They are awful from both a code *and* design POV. Space is a vast expanse and it should be scary to navigate. You shouldnt be able to just move around it freely. 

**b-bbut i-improve dun rem-move!!11**

These have been in the code for **years** with zero improvements being made to them, apart from when we are ripping old systems out to make way for new ones (see: global iterators in #13505).

They are going. Please file your complaints in the neareast shredder. 

## Images of changes
Please see MapDiffBot & IconDiffBot

## Changelog
:cl: AffectedArc07
del: Removed spacepods. If you had purchased Mechanic or Pod Pilot from the karma store, you can get it refunded on the karma refunds tab.
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
CI will almost certainly fail due to some map templates using pods

**Comment by HMBGERDO** [comment]
What about jumpboots and extinguishers? Who does really exploring space in space pod

**Comment by AffectedArc07** [comment]
Also I will take any advice on how to improve the mapping I did here, since its not great.

Obligatory pings to @S34NW and @FreeStylaLT

**Comment by ppi13** [comment]
Ill miss you mechanic and pod pilot.  
I somewhat disagree with the removal, but nothing been done about pods, infact, everything done has been gutting it, removing it's ability to explode, the ability to float in space, nerfed their speed and so on.  
Overall, I will see this as a general good thing for the game.

**Comment by PacifistDalek** [comment]
:/ The only thing I really dislike about this is that I've had some lovely RP time exploring space as a duo with random civilians and the lile. Can there be an increase in accessible space suits or similar gear for space exploring to make up for this?

**Comment by PacifistDalek** [comment]
Actually, here's a question: would creating a thruster module for mechs to make them space manuverable be a reasonable idea, or would that be just replicating the spacepod problem again?

**Comment by S34NW** [comment]
Meta:
image
This area (drone fab) can be moved down to be flush, that extra area that used to function as the engi pod doorway is dead space now.

Delta:
image
This void feels super awkward and un-needed, the solars above can be moved down to create less of a dead corridor.

Cyberiad:
image
Quite frankly this section could do with being all maints now, that long corridor to nowhere next to ass. line is really weird and confusing. I'd make maints encompass the whole area, and widen some of the passages to be at least 2 tiles wide, 1 tile wide maint corridors create problems for player navigation on paradise due to move speed.

**Comment by Wojz** [comment]
It's sad that no one ever wanted to do anything cool about these, I like the jobs conceptually, even if space pods are hot garbage.

**Comment by FlattestGuitar** [comment]
harsh, but fair.

**Comment by AffectedArc07** [comment]
I am now using a new maint for the cyberiad mechanics office, done by s34n

**Comment by S34NW** [comment]
(oops I gave AA the wrong file)

**Comment by Mitchs98** [comment]
Definitely really sad to see the space pods go, especially Space Pod Pilot, especially after I learned you can shoot down the combat drones for loot to give to science. It's going to make dealing with off-station cult bases even harder now too and exploring space more annoying. But my main concern lies with the cult base situation.

Can say I understand their removal due to the shit-code though, even if they still performed alright imo in-game.

**Comment by Spacemanspark** [comment]
Unfortunate, but little to be done now.

**Review by variableundefined — APPROVED** [review]

**Comment by procdrone** [comment]
kill pods, and eventually replace with space mechs

**Comment by tsp01** [comment]
Since the security pod pilot usually acts as a normal security officer, should the maximum amount of security officers be increased by 1?

File: code/__DEFINES/machines.dm [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #define	AUTOLATHE		4	//Uses glass/metal only.
 #define CRAFTLATHE		8	//Uses fuck if I know. For use eventually.
 #define MECHFAB			16 	//Remember, objects utilising this flag should have construction_time and construction_cost vars.
-#define PODFAB			32 	//Used by the spacepod part fabricator. Same idea as the mechfab
+// #define PODFAB			32 	//Used by the spacepod part fabricator. Same idea as the mechfab // AA 2021-10-02 - Removed. Kept for flag consistency.
```
Comment (Fox-McCloud) [inline code/__DEFINES/machines.dm:28]: Probably not necessary in this case, since this isn't a toggle or anything like that.

**Comment by AffectedArc07** [comment]
> Since the security pod pilot usually acts as a normal security officer, should the maximum amount of security officers be increased by 1?

Security officer slots already scale with playercount. 4 civilians = 1 officer slot opened, or vice versa.


If we need extra slots, we have the configuration repo for a reason.

**Comment by Fox-McCloud** [comment]
Good riddance.

**Pull request merged**